andela/might-guy-cfh
# #141409271 CZARs should be able to draw cards by clicking

#### What does this PR do?
- Adds a dialog display to the game page that displays to Czar whenever a new game starts so that the Czar can click to start the new game
- Adds a watch method to the gameController to listen to state changes a especially when  the game state is expecting the Czar to draw card to start a new game
- Adds a method shuffleCard to gameController which is called when Czar clicks to shuffle or draw card, the method closes the dialog and calls the startNextRound method which starts the games next round and update the state of the game.
- Modifies gameUpdate socket emit listener to check for game state of pick black cards that is emitted anytime a new round starts
- Modifies socket.js and game.js to listen to when Czar draws or shuffle cards, execute the appropriate action and emits the update.

#### Description of Task to be completed?
- As a CZAR I should be prompted to click to select from the deck of black cards the next question card.

#### How should this be manually tested?
- Start a new game and at least 3 players join the game. When there are at least 3 players, click the start new game button.
- The current czar sees a dialog box which will prompt him to click to select from the deck of cards and the time will be paused as the other players wait for the czar to click. They will be prompted to wait for the Czar to click card.
- When the Czar clicks the card on the dialog, the next round starts.
- When another round starts and a new car is selected, he is expected to also click the dialog that prompts him to click to shuffle card. The process continues until the end of the game.

#### Any background context you want to provide?
- Before the task, Czar does not have the opportunity to control the game as the question displays automatically. The task solves 2 problem; A Czar knows when he being selected to be the Czar and this feature enables the Czar to have control over the game.

#### What are the relevant pivotal tracker stories?
- CZARs should be able to draw cards by clicking

#### Screenshots (if appropriate)

- ![CZar Shuffle cards dialog](https://raw.githubusercontent.com/andela/might-guy-cfh/feature/141409271/CZARS-draw-cards-clicking/public/img/czar_screenshot.png)

- ![Player waiting for Czar Screen](https://raw.githubusercontent.com/andela/might-guy-cfh/feature/141409271/CZARS-draw-cards-clicking/public/img/player_screenshot.png)

#### Questions:
- NA
